src/profiles/ro_crate/ro_crate_parser.py

```diff
@@ -0,0 +1,407 @@
+
+        Returns:
+            IngestibleDataclasses: the ingestible dataclasses now updated with the datafile
+        """
+        datafile_dict: dict[str, Any] = {}
+        filepath = self.crate.source / filename
+        datafile_dict["filename"] = filename.as_posix()
+        datafile_dict["directory"] = filepath.relative_to(self.crate.source).as_posix()
+        datafile_dict["md5sum"] = checksums.calculate_md5(filepath)
+        mtype, _ = mimetypes.guess_type(filepath)
+        if not mtype:
+            mtype = str(filepath).rsplit(".", maxsplit=1)[-1]
+        datafile_dict["mimetype"] = mtype
+        datafile_dict["size"] = str(filepath.stat().st_size)
+        if rocrate_entity:
+            rocrate_dict: dict[str, Any] = {
+                (self.mapper.get_mt_field(MyTardisObject.DATAFILE, key) or key): value
+                for key, value in rocrate_entity.as_jsonld().items()
+            }
+            datafile_dict.update(rocrate_dict)
+            if datafile_meta := datafile_dict.get("metadata"):
+                datafile_dict["metadata"] = self._read_metadata([datafile_meta])
+        datafile_dict["dataset"] = parent_dataset_description
+        print(datafile_dict["filename"], "my parent is,", datafile_dict["dataset"])
+        raw_datafile: RawDatafile = RawDatafile.model_validate(datafile_dict)
+        raw_datafile.object_schema = self.schema_config.datafile
+        return raw_datafile
+
+    def _process_dataset(self, crate_dataset: DataEntity) -> RawDataset:
+        """Load a dataset and all of it's dataset and datafile children from an RO-Crate spec.
+
+        Recursively load any datasets found as children of this dataset
+        Then parse all datafiles found on disk not already found as part of this or another dataset.
+
+        Args:
+            crate_dataset (DataEntity): roCrate data entity representing this dataset
+            ingestible_classes (IngestibleDataclasses): mytardis dataclasses for datafiles/sets
+
+        Raises:
+            ValueError: (to be updated to more apropriate error type)
+            raised if the roCrate doesn't link this dataset to an experiment/data catalog
+
+        Returns:
+            IngestibleDataclasses:
+            the dataclasses object updated with this dataset and all of it's datafile children
+        """
+        dataset_dict = {
+            (self.mapper.get_mt_field(MyTardisObject.DATASET, key) or key): value
+            for key, value in crate_dataset.as_jsonld().items()
+        }
+
+        dataset_dict["experiments"] = get_norm_value(
+            crate_dataset.as_jsonld(), "includedInDataCatalog"
+        )
+        if len(dataset_dict["experiments"]) == 0:
+            raise ValueError(
+                "Dataset "
+                + str(crate_dataset.id)
+                + " not included in a DataCatalog, cannot link this dataset to experiments"
+            )
```

Comment:
As this relates to a particular Experiment, it's better to either use `logger.error()`, or catch the error and print out a error logging message, so the script doesn't stop running on one invalid Experiment.

Reply:
changed to a logger.error()